src/main/java/ai/knowly/langtorch/connector/pdf/PDFConnectorOption.java

```diff
@@ -4,9 +4,12 @@
 import lombok.Builder;
 import lombok.Data;
 
+import java.util.Optional;
+
 /** Implementation of ReadOption for PDF files. */
 @Data
 @Builder(toBuilder = true, setterPrefix = "set")
 public class PDFConnectorOption implements ConnectorOption {
-  private String filePath;
+  @Builder.Default private Optional<String> filePath = Optional.empty();
+  @Builder.Default private Optional<byte[]> bytes = Optional.empty();
```

Comment:
Let's do the following:

```
private String filePath;
private byte[] fileBytes;

public Optional<String> getFilePath() {
    Optional.ofNullable(filePath);
}
public Optional<byte[]> getFilePath() {}

```

If you are making an optional field inside, you must set an optional one. but we never set an empty value in the builder. we only set nonnull values. Optional wont matter until we read it.